teamcity: failed tests on release-1.1: testrace/TestRegistryResumeActiveLease

The following tests appear to have failed:

[#362264](https://teamcity.cockroachdb.com/viewLog.html?buildId=362264):

```
--- FAIL: testrace/TestRegistryResumeActiveLease (0.000s)
Race detected!

------- Stdout: -------
I170927 02:41:21.824002 5639 server/config.go:528  [n?] 1 storage engine initialized
I170927 02:41:21.824422 5639 server/config.go:530  [n?] RocksDB cache size: 128 MiB
I170927 02:41:21.824457 5639 server/config.go:530  [n?] store 0: in-memory, size 0 B
I170927 02:41:21.879175 5639 server/node.go:344  [n?] **** cluster 7bdbbc5d-1e32-48a3-b5db-532af1e193d1 has been created
I170927 02:41:21.879256 5639 server/server.go:835  [n?] **** add additional nodes by specifying --join=127.0.0.1:47672
I170927 02:41:21.901701 5639 storage/store.go:1199  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170927 02:41:21.907216 5639 server/node.go:461  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=3.2 KiB), ranges=1, leases=1, writes=0.00, bytesPerReplica={p10=3266.00 p25=3266.00 p50=3266.00 p75=3266.00 p90=3266.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I170927 02:41:21.907427 5639 server/node.go:326  [n1] node ID 1 initialized
I170927 02:41:21.907702 5639 gossip/gossip.go:327  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:47672" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:0 patch:0 unstable:3 > 
I170927 02:41:21.908078 5639 storage/stores.go:303  [n1] read 0 node addresses from persistent storage
I170927 02:41:21.908349 5639 server/node.go:606  [n1] connecting to gossip network to verify cluster ID...
I170927 02:41:21.911074 5639 server/node.go:631  [n1] node connected via gossip and verified as part of cluster "7bdbbc5d-1e32-48a3-b5db-532af1e193d1"
I170927 02:41:21.911396 5639 server/node.go:403  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I170927 02:41:21.916209 5741 storage/replica_command.go:2710  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I170927 02:41:21.934998 5639 sql/executor.go:408  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:47672}
I170927 02:41:21.987688 5741 storage/replica_command.go:2710  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
I170927 02:41:22.037615 5741 storage/replica_command.go:2710  [split,n1,s1,r3/1:/{System/NodeL���-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I170927 02:41:22.062445 5639 server/server.go:927  [n1] starting https server at 127.0.0.1:59417
I170927 02:41:22.062524 5639 server/server.go:928  [n1] starting grpc/postgres server at 127.0.0.1:47672
I170927 02:41:22.062568 5639 server/server.go:929  [n1] advertising CockroachDB node at 127.0.0.1:47672
W170927 02:41:22.062836 5639 sql/jobs/registry.go:156  [n1] unable to get node liveness: node not in the liveness table
I170927 02:41:22.167303 5639 sql/event_log.go:102  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
E170927 02:41:22.291943 5851 storage/replica_proposal.go:538  [n1,s1,r3/1:/{System/NodeL���-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I170927 02:41:22.296165 5741 storage/replica_command.go:2710  [split,n1,s1,r4/1:/{System/NodeL���-Max}] initiating a split of this range at key /System/tsd [r5]
W170927 02:41:22.298530 5999 storage/intent_resolver.go:332  [n1,s1,r4/1:/{System/NodeL���-Max}]: failed to push during intent resolution: failed to push "unnamed" id=6c2adbc4 key=/Table/SystemConfigSpan/Start rw=true pri=0.03078924 iso=SERIALIZABLE stat=PENDING epo=0 ts=1506480082.168053663,0 orig=1506480082.168053663,0 max=1506480082.168053663,0 wto=false rop=false seq=3
E170927 02:41:22.300461 6022 sql/jobs/registry.go:143  error while adopting jobs: relation "system.jobs" does not exist
I170927 02:41:22.338251 5639 sql/lease.go:341  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-09-27 02:41:22.337090701 +0000 UTC
I170927 02:41:22.371815 5741 storage/replica_command.go:2710  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I170927 02:41:22.434149 5741 storage/replica_command.go:2710  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
E170927 02:41:22.437134 6022 sql/jobs/registry.go:143  error while adopting jobs: relation "system.jobs" does not exist
E170927 02:41:22.469442 5724 storage/replica_proposal.go:538  [n1,s1,r6/1:/{System/tse-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I170927 02:41:22.474732 5741 storage/replica_command.go:2710  [split,n1,s1,r7/1:/{Table/System���-Max}] initiating a split of this range at key /Table/11 [r8]
W170927 02:41:22.484687 6134 storage/intent_resolver.go:332  [n1,s1,r7/1:/{Table/System���-Max}]: failed to push during intent resolution: failed to push "unnamed" id=cc3abd31 key=/Table/SystemConfigSpan/Start rw=true pri=0.01250627 iso=SERIALIZABLE stat=PENDING epo=0 ts=1506480082.380993627,0 orig=1506480082.380993627,0 max=1506480082.380993627,0 wto=false rop=false seq=4
I170927 02:41:22.574296 5741 storage/replica_command.go:2710  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I170927 02:41:22.627489 5639 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I170927 02:41:22.652045 5741 storage/replica_command.go:2710  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I170927 02:41:22.739923 5741 storage/replica_command.go:2710  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I170927 02:41:22.775514 5639 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:1.0-3 User:node}
I170927 02:41:22.858722 5741 storage/replica_command.go:2710  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I170927 02:41:22.874280 5639 sql/event_log.go:102  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I170927 02:41:22.920933 5639 server/server.go:1071  [n1] done ensuring all necessary migrations have run
I170927 02:41:22.921042 5639 server/server.go:1073  [n1] serving sql connections
I170927 02:41:22.936262 5741 storage/replica_command.go:2710  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I170927 02:41:22.945657 6352 sql/event_log.go:102  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:47672} Attrs: Locality: ServerVersion:1.0-3} ClusterID:7bdbbc5d-1e32-48a3-b5db-532af1e193d1 StartedAt:1506480081911167054 LastUp:1506480081911167054}
I170927 02:41:22.985356 5741 storage/replica_command.go:2710  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I170927 02:41:23.048643 5741 storage/replica_command.go:2710  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I170927 02:41:23.105480 5741 storage/replica_command.go:2710  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I170927 02:41:23.434507 6662 sql/jobs/registry.go:293  job 283147250270175233: resuming
I170927 02:41:23.436418 5912 kv/transport_race.go:69  transport race promotion: ran 37 iterations on up to 508 requests
==================
Read at 0x00c42104602c by goroutine 121:
  internal/race.Read()
      /usr/local/go/src/internal/race/race.go:37 +0x38
  sync.(*WaitGroup).Add()
      /usr/local/go/src/sync/waitgroup.go:71 +0x26b
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:190 +0x84
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:106 +0x33f
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1123 +0x3a7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:382 +0x352
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:446 +0x1b7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:930 +0x574
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:795 +0xd1d
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:597 +0x3ed
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:444 +0x281
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:548 +0x209
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:952 +0x5b2
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Send-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:510 +0xa3
  github.com/cockroachdb/cockroach/pkg/internal/client.sendAndFill()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:454 +0x155
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Run()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:510 +0x111
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Get()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:361 +0x10a
  github.com/cockroachdb/cockroach/pkg/sql.getDescriptor()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/descriptor.go:172 +0xdc
  github.com/cockroachdb/cockroach/pkg/sql.getDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:125 +0x173
  github.com/cockroachdb/cockroach/pkg/sql.MustGetDatabaseDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/database.go:151 +0x93
  github.com/cockroachdb/cockroach/pkg/sql.getTableOrViewDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:111 +0x19a
  github.com/cockroachdb/cockroach/pkg/sql.getTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:136 +0x80
  github.com/cockroachdb/cockroach/pkg/sql.MustGetTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:191 +0x91
  github.com/cockroachdb/cockroach/pkg/sql.(*TableCollection).getTableVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/table.go:307 +0x17e
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getTableDesc()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:548 +0x1f5
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getTableScanOrViewPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:529 +0x98
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:351 +0x13ba
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getDataSource()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:394 +0x644
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).getSources()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/data_source.go:253 +0x485
  github.com/cockroachdb/cockroach/pkg/sql.(*renderNode).initFrom()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:302 +0x12c
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).SelectClause()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:219 +0x10d
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).Select()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:163 +0x98b
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).newPlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:394 +0x4401
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).makePlan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:187 +0xa2
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).query()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:233 +0x1c0
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).queryRows()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:258 +0xe7
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:240 +0xab
  github.com/cockroachdb/cockroach/pkg/sql.InternalExecutor.QueryRowInTransaction()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:59 +0x183
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).update.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:507 +0x1ee
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:510 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:754 +0xc5
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:511 +0x17c
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).runInTxn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:436 +0x208
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).update()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:546 +0x1a4
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).Succeeded()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:325 +0xd8
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Job).FinishedWith()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/jobs.go:374 +0x25a
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).maybeAdoptJob.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:313 +0x1bb

Previous write at 0x00c42104602c by goroutine 27:
  internal/race.Write()
      /usr/local/go/src/internal/race/race.go:41 +0x38
  sync.(*WaitGroup).Wait()
      /usr/local/go/src/sync/waitgroup.go:129 +0x14b
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:423 +0x2d2
  github.com/cockroachdb/cockroach/pkg/sql/jobs_test.TestRegistryResumeActiveLease()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry_external_test.go:212 +0x91f
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107

Goroutine 121 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).maybeAdoptJob()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:317 +0xbdb
  github.com/cockroachdb/cockroach/pkg/sql/jobs.(*Registry).Start.func2()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/registry.go:142 +0x154
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x160

Goroutine 27 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107
  testing.runTests()
      /usr/local/go/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:822 +0x1c3
  github.com/cockroachdb/cockroach/pkg/sql/jobs_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/jobs/main_test.go:33 +0x189
  main.main()
      github.com/cockroachdb/cockroach/pkg/sql/jobs/_test/_testmain.go:54 +0x20f
==================
I170927 02:41:11.019045 1 rand.go:74  Random seed: -9156376665017291678

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/kv/transport_race.go
+++ b/pkg/kv/transport_race.go
@@ -59,53 +59,57 @@ func GRPCTransportFactory(
 	if atomic.AddInt32(&running, 1) <= 1 {
 		// NB: We can't use Stopper.RunWorker because doing so would race with
 		// calling Stopper.Stop.
-		rpcContext.Stopper.RunAsyncTask(context.TODO(), "transport racer", func(ctx context.Context) {
-			var iters int
-			var curIdx int
-			defer func() {
-				atomic.StoreInt32(&running, 0)
-				log.Infof(
-					ctx,
-					"transport race promotion: ran %d iterations on up to %d requests",
-					iters, curIdx+1,
-				)
-			}()
-			// Make a fixed-size slice of *BatchRequest. When full, entries
-			// are evicted in FIFO order.
-			const size = 1000
-			bas := make([]*roachpb.BatchRequest, size)
-			encoder := gob.NewEncoder(ioutil.Discard)
-			for {
-				iters++
-				start := timeutil.Now()
-				for _, ba := range bas {
-					if ba != nil {
-						if err := encoder.Encode(ba); err != nil {
-							panic(err)
+		if err := rpcContext.Stopper.RunAsyncTask(
+			context.TODO(), "transport racer", func(ctx context.Context) {
+				var iters int
+				var curIdx int
+				defer func() {
+					atomic.StoreInt32(&running, 0)
+					log.Infof(
+						ctx,
+						"transport race promotion: ran %d iterations on up to %d requests",
+						iters, curIdx+1,
+					)
+				}()
+				// Make a fixed-size slice of *BatchRequest. When full, entries
+				// are evicted in FIFO order.
+				const size = 1000
+				bas := make([]*roachpb.BatchRequest, size)
+				encoder := gob.NewEncoder(ioutil.Discard)
+				for {
+					iters++
+					start := timeutil.Now()
+					for _, ba := range bas {
+						if ba != nil {
+							if err := encoder.Encode(ba); err != nil {
+								panic(err)
+							}
 						}
 					}
-				}
-				// Prevent the goroutine from spinning too hot as this lets CI
-				// times skyrocket. Sleep on average for as long as we worked
-				// on the last iteration so we spend no more than half our CPU
-				// time on this task.
-				jittered := time.After(jitter(timeutil.Since(start)))
-				// Collect incoming requests until the jittered timer fires,
-				// then access everything we have.
-				for {
-					select {
-					case <-rpcContext.Stopper.ShouldQuiesce():
-						return
-					case ba := <-incoming:
-						bas[curIdx%size] = ba
-						curIdx++
-						continue
-					case <-jittered:
+					// Prevent the goroutine from spinning too hot as this lets CI
+					// times skyrocket. Sleep on average for as long as we worked
+					// on the last iteration so we spend no more than half our CPU
+					// time on this task.
+					jittered := time.After(jitter(timeutil.Since(start)))
+					// Collect incoming requests until the jittered timer fires,
+					// then access everything we have.
+					for {
+						select {
+						case <-rpcContext.Stopper.ShouldQuiesce():
+							return
+						case ba := <-incoming:
+							bas[curIdx%size] = ba
+							curIdx++
+							continue
+						case <-jittered:
+						}
+						break
 					}
-					break
 				}
-			}
-		})
+			}); err != nil {
+			// Failed to start async task, reset our state.
+			atomic.StoreInt32(&running, 0)
+		}
 	}
 	select {
 	// We have a shallow copy here and so the top level scalar fields can't

--- a/pkg/kv/transport_race.go
+++ b/pkg/kv/transport_race.go
@@ -57,7 +57,9 @@ func GRPCTransportFactory(
 	opts SendOptions, rpcContext *rpc.Context, replicas ReplicaSlice, args roachpb.BatchRequest,
 ) (Transport, error) {
 	if atomic.AddInt32(&running, 1) <= 1 {
-		rpcContext.Stopper.RunWorker(context.TODO(), func(ctx context.Context) {
+		// NB: We can't use Stopper.RunWorker because doing so would race with
+		// calling Stopper.Stop.
+		rpcContext.Stopper.RunAsyncTask(context.TODO(), "transport racer", func(ctx context.Context) {
 			var iters int
 			var curIdx int
 			defer func() {
@@ -92,7 +94,7 @@ func GRPCTransportFactory(
 				// then access everything we have.
 				for {
 					select {
-					case <-rpcContext.Stopper.ShouldStop():
+					case <-rpcContext.Stopper.ShouldQuiesce():
 						return
 					case ba := <-incoming:
 						bas[curIdx%size] = ba

